### DISCUSSION TITLE
not found /manifest.json 404

### DISCUSSION THREAD

**Discussion by algorithmlover2016** [open-webui/open-webui#1683] [General]
# Bug Report

## Description
NotFound [Error]: Not found: /manifest.json
    at resolve (/disks/ssd2/litgao/codes/open-webui/node_modules/@sveltejs/kit/src/runtime/server/respond.js:483:13)
    at resolve (/disks/ssd2/litgao/codes/open-webui/node_modules/@sveltejs/kit/src/runtime/server/respond.js:277:5)
    at #options.hooks.handle (/disks/ssd2/litgao/codes/open-webui/node_modules/@sveltejs/kit/src/runtime/server/index.js:49:56)
    at Module.respond (/disks/ssd2/litgao/codes/open-webui/node_modules/@sveltejs/kit/src/runtime/server/respond.js:274:40)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  status: 404
}
**Bug Summary:**
Not found: /manifest.json

**Steps to Reproduce:**
cd open-webui && npn install && npm run build && npm run dev

## Environment

- **Open WebUI Version:**  main 
- **Ollama (if applicable):** [e.g., 0.1.30, 0.1.32-rc1]

- **Operating System:** [e.g., Windows 10, macOS Big Sur, Ubuntu 20.04]
- **Browser (if applicable):** [e.g., Chrome 100.0, Firefox 98.0]

## Reproduction Details

**Confirmation:**

## Logs and Screenshots
![image](https://github.com/open-webui/open-webui/assets/17782711/b56c6656-04e9-4d59-b38f-e3b78f91931b)

**Comment by algorithmlover2016** [comment]
https://github.com/open-webui/open-webui/commit/83ad488e314dff5c0eceb700221c4374229b31e9

**Comment by algorithmlover2016** [comment]
I find that it need load static/manifest.json file, which means it will visit http://localhost:5173/manifest.json
It will not visit to backend's server, which is http://localhost:8080/manifest.json.

What should do to make dynamically load manifest.json successfully for src/app.html.
<link rel="manifest" href="%sveltekit.assets%/manifest.json" crossorigin="use-credentials" />

**Comment by justinh-rahb** [comment]
I'm not quite understanding what the problem is here, more explanation may help.

**Reply by algorithmlover2016** [reply]
I mean the ui can not find the manifest.json, which mean that localhost:8080/manifest.json is not found in the main branch.

**Reply by algorithmlover2016** [reply]
![image](https://github.com/open-webui/open-webui/assets/17782711/3ac1b6c1-ad26-4b3d-a45e-2e876510cf78)
In this html file, it will visit %sveltekit.assets%/manifest.json which should be located in static/manifest.json. It will not try to make a get request to the backend.
![image](https://github.com/open-webui/open-webui/assets/17782711/c54870e1-27cb-43d5-9bbc-1192f44061df)

**Reply by algorithmlover2016** [reply]
After all, if I did not place manifest.json in static/ folder, it will warns me that "Not Found /manifest.json status 404". And if I place manifest.json in static/ folder, the warning disappears.

**Reply by justinh-rahb** [reply]
Thank you for highlighting this concern. This warning is specific to the development environment and does not impact the production deployed app's functionality. The `manifest.json` file will be correctly generated and available in the production environment. Hence, we can consider this behavior expected during development and not a cause for concern.

**Comment by WHG555** [comment]
Excuse me, how to start open-webui with npm run dev

**Comment by WHG555** [comment]
![image](https://github.com/open-webui/open-webui/assets/26180549/6ba529d0-5d1f-4b23-a1ab-640492ccee73)

**Reply by algorithmlover2016** [reply]
you can ignore this error message about no found /manifest.json.
If you have to fix this issue, just create a manifest.json in static/ folder. 
<img width="767" alt="image" src="https://github.com/open-webui/open-webui/assets/17782711/f0c61700-3a05-4ac5-9d5a-570c21a32574">
here is the content of the json file:
`{
	"name": "Open WebUI",
	"short_name": "Open WebUI",
	"start_url": "/",
	"display": "standalone",
	"background_color": "#343541",
	"theme_color": "#343541",
	"orientation": "portrait-primary",
	"icons": [
		{
			"src": "/favicon.png",
			"type": "image/png",
			"sizes": "844x884"
		}
	]
}`
